firstvoices/backend/serializers/dictionary_serializers.py

```diff
@@ -69,31 +87,151 @@ class Meta(SiteContentLinkedTitleSerializer.Meta):
 
 
 class DictionaryEntryDetailSerializer(
-    RelatedMediaSerializerMixin, serializers.HyperlinkedModelSerializer
+    CreateSiteContentSerializerMixin,
+    RelatedMediaSerializerMixin,
+    serializers.HyperlinkedModelSerializer,
+    UpdateSerializerMixin,
 ):
-    url = SiteHyperlinkedIdentityField(view_name="api:dictionaryentry-detail")
-    visibility = serializers.CharField(source="get_visibility_display")
-    translations = TranslationSerializer(source="translation_set", many=True)
-    pronunciations = PronunciationSerializer(source="pronunciation_set", many=True)
-    notes = NoteSerializer(source="note_set", many=True)
+    url = SiteHyperlinkedIdentityField(
+        read_only=True, view_name="api:dictionaryentry-detail"
+    )
+    type = serializers.ChoiceField(
+        allow_null=False,
+        choices=dictionary.TypeOfDictionaryEntry.choices,
+        default=dictionary.TypeOfDictionaryEntry.WORD,
+    )
+    custom_order = serializers.CharField(read_only=True)
+    visibility = serializers.CharField(read_only=True, source="get_visibility_display")
+    visibility_value = serializers.ChoiceField(
+        choices=Visibility.choices, default=Visibility.TEAM, write_only=True
+    )
+    categories = WritableCategorySerializer(
+        queryset=category.Category.objects.all(),
+        many=True,
+        required=False,
+    )
     acknowledgements = AcknowledgementSerializer(
-        source="acknowledgement_set", many=True
+        many=True,
+        required=False,
+        source="acknowledgement_set",
     )
     alternate_spellings = AlternateSpellingSerializer(
-        source="alternatespelling_set", many=True
+        many=True, required=False, source="alternatespelling_set"
     )
-    categories = CategorySerializer(many=True)
-    site = LinkedSiteSerializer()
+    notes = NoteSerializer(many=True, required=False, source="note_set")
+    translations = TranslationSerializer(
+        many=True, required=False, source="translation_set"
+    )
+    pronunciations = PronunciationSerializer(
+        many=True, required=False, source="pronunciation_set"
+    )
+
+    site = LinkedSiteSerializer(read_only=True, required=False)
     related_entries = DictionaryEntrySummarySerializer(
-        source="related_dictionary_entries", many=True
+        many=True,
+        required=False,
+        source="related_dictionary_entries",
     )
-    split_chars = serializers.SerializerMethodField()
-    split_chars_base = serializers.SerializerMethodField()
-    split_words = serializers.SerializerMethodField()
-    split_words_base = serializers.SerializerMethodField()
+    split_chars = serializers.SerializerMethodField(read_only=True)
+    split_chars_base = serializers.SerializerMethodField(read_only=True)
+    split_words = serializers.SerializerMethodField(read_only=True)
+    split_words_base = serializers.SerializerMethodField(read_only=True)
 
     logger = logging.getLogger(__name__)
 
+    def validate(self, attrs):
```

Comment:
just curious, we have a SameSite validator, can that be used here as well by adding to the categories field ? or do we need object level validation instead of the field level validation ?

Reply:
I tested using the samesite validator here and it was not being applied, similar to related dictionary entries.